lib/models/channel/channel.dart

```diff
@@ -1 +1,55 @@
-class Channel {}
+import 'package:json_annotation/json_annotation.dart';
+import 'package:twake/models/channel/channel_visibility.dart';
+
+part 'channel.g.dart';
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class Channel {
+  final String id;
+
+  final String name;
+
+  final String? icon;
+
+  final String? description;
+
+  final String companyId;
+
+  final String workspaceId;
+
+  final int membersCount;
+
+  final List<String> members;
+
+  final ChannelVisibility visibility;
+
+  final int lastActivity;
+
+  final int userLastAccess;
+
+  final String? draft;
+
+  List<String> permissions;
+
+  bool get hasUnread => userLastAccess < lastActivity;
+
+  Channel({
+    required this.id,
+    required this.name,
+    this.icon,
+    this.description,
+    required this.companyId,
+    required this.workspaceId,
+    required this.membersCount,
+    required this.members,
+    required this.visibility,
+    required this.lastActivity,
+    required this.userLastAccess,
+    this.draft,
+    required this.permissions
+  });
+
+  factory Channel.fromJson(Map<String, dynamic> json) => _$ChannelFromJson(json);
```

Comment:
We have a problem here, it's related to SQLite datatypes, and there's only four of them: TEXT, INT, REAL, BLOB. 
And List<T> and Map<K,V> is not one of them. So if we were to store those model to database will get errors. 
So I suggest to add extra logic to convert those types to string and back during the storage and retrieval from DB. 
You can refer to Message model for reference as usual.

Reply:
https://github.com/linagora/Twake-Mobile/blob/V3/lib/models/message/message.dart

Reply:
@bobs4462 done